initdb.py

```diff
@@ -0,0 +1,140 @@
+# Initializes a MySQL Database
+# 1) connect to server
+# 2) create database with cursor
+# 3) connect to database
+# 4) create tables with cursor
+# 5) commit transactoins
+# 6) exit
+
+
+
+
+import sys
+import mysql.connector
+from mysql.connector import errorcode
+
+# Define constants
+DB_NAME = "starbucksdb"
+
+TABLES = {}
+# Create table statments
+# replace with own schema
+
+# https://www.kaggle.com/starbucks/store-locations
+TABLES['locations']=(
+    "CREATE TABLE `locations` ("
+    "   `brand` varchar(15) NOT NULL,"
+    "   `store_number` varchar(20) NOT NULL,"
+    "   `store_name` varchar(15) NOT NULL,"
+    "   `ownership_type` varchar(10) NOT NULL,"
+    "   `street_address` varchar(25) NOT NULL,"
+    "   `city` varchar(15) NOT NULL,"
+    "   `state` varchar(2) NOT NULL,"
+    "   `country` varchar(2) NOT NULL,"
+    "   `postcode` varchar(10),"
+    "   `phone_number` varchar(11),"
+    "   `timezone` varchar(20) NOT NULL,"
+    "   `longitude` decimal(4,2) NOT NULL,"
+    "   `latitude` decimal(4,2) NOT NULL"
+    ") ENGINE=InnoDB")
+
+# https://github.com/kdallas2/diversity/blob/master/di.csv
+TABLES['diversity'] = (
+    "CREATE TABLE `diversity` ("
+    "   `location` varchar(15) NOT NULL,"
+    "   `diversity_index` decimal(7,6) NOT NULL,"
+    "   `african_american` decimal(4,2) NOT NULL,"
+    "   `american_indian` decimal(4,2) NOT NULL,"
+    "   `asian` decimal(4,2) NOT NULL,"
+    "   `pacific_islander` decimal(4,2) NOT NULL,"
+    "   `two_or_more` decimal(4,2) NOT NULL,"
+    "   `hispanic_latino` decimal(4,2) NOT NULL,"
+    "   `white_only` decimal(4,2) NOT NULL"
+    ") ENGINE=InnoDB")
+
+
+
+
+def create_database(cursor, connection):
+    """
+    Attempt to create the DB_NAME database
+    """
+    try:
+       create_sda_db = "CREATE DATABASE {}".format(DB_NAME)
+       print("Creating database {}".format(DB_NAME))
+       cursor.execute(create_sda_db)
+    except mysql.connector.Error as mysqlError:
+        print("Failed creating database: {}".format(mysqlError))
+        sys.exit()
+    # commit transactions
+    connection.commit()
+
+def connect_to_database(connection):
+    """
+    Attempt to connecect to DB_NAME database
+    """
+    try:
+        connection.database = DB_NAME
+    except mysql.connector.Error as mysqlError:
+        if mysqlError.errno == errorcode.ER_BAD_DB_ERROR:
+            print("Database does not exist")
+        else:
+            print(mysqlError)
+            sys.exit()
+
+def create_tables(cursor, connection):
+    """
+    Attempt to execute each CREATE statement in TABLES
+    to create tables.
+    NOTE: MUST BE CONNECTED TO A DB FIRST
```

Comment:
The ````connnect_database()```` function throws an error and exits if it can't connect to a database. By the time ````create_tables()```` is called, the connection to a database has either been made or the program has exited with an error. But I am going to move connecting to the database inside this function!